### PR TITLE
[1LP][RFR]Automate test: test_custom_logos_via_api AND add widgets for CustomLo…

### PR DESCRIPTION
**Pull request by valaparthvi**
Changes introduced with this PR:
1. Automate test: test_custom_logos_via_api
2. Add missing widgets for `CustomLogos`.

{{ pytest: cfme/tests/test_rest.py::test_custom_logos_via_api --use-template-cache -sqvvv }}

PS: I feel the need to appreciate @digitronik's effort in helping me with this PR. Thanks Nikhil :)

### CODE REVIEW

File: cfme/base/__init__.py [hunk 1]
```diff
@@ -105,6 +105,27 @@ def update_advanced_settings(self, settings_dict):
...
+        view = navigate_to(self, "CustomLogos")
+        logo_view = getattr(view.customlogos, file_type)
+        if file_data:
+            if file_type == "logintext":
+                logo_view.fill({"login_text": file_data})
+            else:
+                logo_view.fill({"image": file_data})
+                logo_view.upload_button.click()
+            view.flash.assert_no_error()
+        logo_view.enable.fill(enable)
```
[the remark this answers is not in the document]
Comment (valaparthvi) [reply cfme/base/__init__.py]: Oh yes! I knew I was missing something. Thanks, Shveta :+1:

File: cfme/base/__init__.py [hunk 2]
```diff
@@ -105,6 +105,29 @@ def update_advanced_settings(self, settings_dict):
...
+        """
+        This function can be used to upload custom logo or text and use them.
+
+        Args:
+            file_type (str) : Can be either of [logo, login_logo, brand, favicon, logintext]
+            file_data (str) : Text data if file_type is logintext else image name to be uploaded
+            enable (bool) : True to use the custom logo/text else False
+        """
+        view = navigate_to(self, "CustomLogos")
+        logo_view = getattr(view.customlogos, file_type)
```
Comment (digitronik) [inline cfme/base/__init__.py]: I would check attribute collected or not else raise `ValueError` means if `file_type` not matched.

File: cfme/base/__init__.py [hunk 3]
```diff
@@ -105,6 +105,29 @@ def update_advanced_settings(self, settings_dict):
         )
         assert result.ok
 
+    def upload_custom_logo(self, file_type, file_data=None, enable=True):
+        """
+        This function can be used to upload custom logo or text and use them.
+
+        Args:
+            file_type (str) : Can be either of [logo, login_logo, brand, favicon, logintext]
+            file_data (str) : Text data if file_type is logintext else image name to be uploaded
```
Comment (digitronik) [inline cfme/base/__init__.py]: I think we need `image path` not only `image name` while uploading logo

File: cfme/base/__init__.py [hunk 4]
```diff
@@ -105,6 +105,29 @@ def update_advanced_settings(self, settings_dict):
...
+        view = navigate_to(self, "CustomLogos")
+        logo_view = getattr(view.customlogos, file_type)
+        if file_data:
+            if file_type == "logintext":
+                logo_view.fill({"login_text": file_data})
+            else:
+                logo_view.fill({"image": file_data})
+                logo_view.upload_button.click()
+            view.flash.assert_no_error()
+        if logo_view.enable.is_displayed:
```
Comment (digitronik) [inline cfme/base/__init__.py]: Q: perform `enable` operation only if `is_displayed`?

Comment (valaparthvi) [reply cfme/base/__init__.py]: Yes, if there is no custom logo present, this button will not be visible. We want to use to this function to upload custom logo and `enable` helps us decide if we want to use that logo. If a logo is already uploaded and used, but we do not want to use it anymore, we would use this function to disable the use by passing `enable=False`.

**Comment by valaparthvi** [comment]
PRT will fail for 5.11, sprout is unable to provision appliances @john-dupuy.

**Pull request merged**